src/controllers/EvidenceDownloadController.ts

```diff
@@ -34,21 +45,44 @@ export class EvidenceDownloadController extends BaseAsyncHttpController {
             return this.renderDownloadError();
         }
 
-        let readable: Readable;
+        return await this.downloadAttachment(metadata, this.httpContext.response);
+    }
+
+    private async downloadAttachment(metadata: FileMetadata, res: Response): Promise<void> {
+
         try {
-            readable = await this.fileTransferService.download(fileId);
+            const downloadStream: Readable = await this.fileTransferService.download(metadata.id);
+            res.setHeader('content-disposition', `attachment; filename=${metadata.name}`);
+            return this.pipeDataIntoStream(downloadStream, res);
+
         } catch (err) {
+
             if (err instanceof FileNotReadyError) {
                 return this.renderDownloadError();
             } else {
                 throw err;
             }
         }
+    }
 
-        this.httpContext.response.setHeader('content-disposition', `attachment; filename=${metadata.name}`);
-        return await this.pipeDataIntoStream(readable, this.httpContext.response);
+    private getDownloadPath(req: Request, fileId: string): string {
+
+        const appealId = req.query[APPEAL_ID_QUERY_KEY];
+        const companyNumber = req.query[COMPANY_NUMBER_QUERY_KEY];
+
+        if (!appealId && !companyNumber) {
```

Comment:
These tests are in LoadAppealMiddleware. This logic should be as simple as throwing an error here. Perhaps we could add some tests in case the user decides to manipulate the urls

Reply:
Yeah i did see that the LoadApealMiddleware checks for these parameters and there are tests covering this. I agree with tests covering this logic incase of manipulation